### PR TITLE
retrieve rows through a method

### PR DESCRIPTION
**Pull request by faustbrian**
Rework of https://github.com/calebporzio/sushi/pull/2 due to https://github.com/calebporzio/sushi/pull/2#issuecomment-580317930.

1. Do we just want to delete all rows instead of doing a full reset of the cache file and thus could assert a count of 0 after the flush call?
2. I couldn't really think of a good way to test https://github.com/calebporzio/sushi/pull/31/files#diff-1424272f1875ed72070adab615309d88R63-R70 due to point 1.

### CODE REVIEW

**Comment by faustbrian** [comment]
Looks like tests will be red until https://github.com/calebporzio/sushi/pull/25 is merged.

File: src/Sushi.php [hunk 1]
```diff
@@ -67,9 +61,10 @@ protected static function setSqliteConnection($database)
 
     public function migrate()
     {
-        throw_unless(is_array($this->rows), new \Exception('Sushi: $rows property not found on model: '.get_class($this)));
+        $rows = $this->getRows();
+
+        throw_unless(is_array($rows), new \Exception('Sushi: $rows property not found on model: '.get_class($this)));
```
Comment (shadoWalker89) [inline src/Sushi.php]: The exception message will be misleading when the developper overrides the `getRows()` method and doesn't return an array, i think it should be updated if you are going to keep the check there or maybe just move the check and the exception to the default `getRows()`

**Comment by shadoWalker89** [comment]
This PR allows to define `getRows()` but doesn't allow for a custom check if the cache is valid or not

https://github.com/calebporzio/sushi/blob/d8844f379e43fe7262e1356f4c8a3a759bc02cf2/src/Sushi.php#L40

The file time check up here should be extracted to a method to allow the developper to override that as well when he overrides the `getRows()`.

**Comment by calebporzio** [comment]
Echoing what @shadoWalker89 said @faustbrian.

This is a good PR, just wondering about the previously discussed `->flushRowCache()` and the corresponding check for its existance if using `->getRows()`.

Sorry if I missed something along the way where you explain your reasoning for leaving them out.

Thanks!

**Comment by faustbrian** [comment]
`flushRowCache` is always defined at https://github.com/calebporzio/sushi/pull/31/files#diff-26e616adb7433e8dcaa674db64d6dddbR69-R74 so I am not sure what check of existence I should be adding for it or did you mean that the method should usually not exist and use `method_exists` to check for it and throw in certain scenarios?

The behaviour at the moment is also still the same with `getRows` at https://github.com/calebporzio/sushi/pull/31/files#diff-26e616adb7433e8dcaa674db64d6dddbR76-R83 where I moved the default `$rows` property based behaviour into it.

**Pull request merged**